### PR TITLE
Restore the invariant that MembersAndInitializers do not have empty groups of sibling initializers.

### LINKED ISSUE #50775: Refine design of SourceMemberContainerSymbol.AddDeclaredNontypeMembers
Follow-up on comment https://github.com/dotnet/roslyn/pull/50650#discussion_r562274378

Relates to the code below. We may not need this array builder.

Tagging @AlekseyTs if you want to add any details.

```csharp
        private void AddDeclaredNontypeMembers(DeclaredMembersAndInitializersBuilder builder, DiagnosticBag diagnostics)
        {
...
            static ArrayBuilder<FieldOrPropertyInitializer>? noteRecordParameters(RecordDeclarationSyntax syntax, DeclaredMembersAndInitializersBuilder builder, DiagnosticBag diagnostics)
            {
                var parameterList = syntax.ParameterList;
                if (parameterList is null)
                {
                    return null;
                }

                if (builder.RecordDeclarationWithParameters is null)
                {
                    builder.RecordDeclarationWithParameters = syntax;
                    // Keep track of which instance initializers we'll want to update when we synthesize record properties
                    builder.InstanceInitializersIndexForRecordDeclarationWithParameters = builder.InstanceInitializers.Count;
                    return ArrayBuilder<FieldOrPropertyInitializer>.GetInstance();
                }
                else
                {
                    diagnostics.Add(ErrorCode.ERR_MultipleRecordParameterLists, parameterList.Location);
                    return null;
                }
            }
        }
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Closes #50775.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@jcouv, @dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@jcouv, @dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@jcouv, @dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@jcouv, @dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs:2481]: Comment can be removed too

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2) modulo out-of-date XML doc comment

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Review by chsienki — APPROVED** [review]
:shipit:

**Pull request merged**
